main/lsp/lsp_helpers.cc

```diff
@@ -1,26 +1,42 @@
+#include "absl/strings/escaping.h"
 #include "absl/strings/match.h"
 #include "absl/strings/str_cat.h"
+#include "absl/strings/str_replace.h"
+#include "common/FileOps.h"
 #include "lsp.h"
 
 using namespace std;
 
 namespace sorbet::realmain::lsp {
 
+constexpr string_view sorbetScheme = "sorbet:";
+constexpr string_view httpsScheme = "https";
+
 string LSPLoop::remoteName2Local(string_view uri) {
-    ENFORCE(absl::StartsWith(uri, rootUri));
-    const char *start = uri.data() + rootUri.length();
+    const bool isSorbetURI = absl::StartsWith(uri, sorbetScheme);
+    ENFORCE(absl::StartsWith(uri, rootUri) || (enableSorbetURIs && isSorbetURI));
```

Comment:
Is this an `ENFORCE` that can be violated by a bad LSP implementation? If so, is there some way of handling this that doesn't rely on an ENFORCE but can report an error?

Reply:
Good point. I'll add an error message and have it return the URI untransformed, which should cause a visible error later on (likely in the form of sending the error "file not found" to the client).